src/storage/txn/process.rs

```diff
@@ -565,44 +613,61 @@ fn process_write_impl<E: Engine>(
         _ => panic!("unsupported write command"),
     };
 
-    box_try!(scheduler.schedule(Msg::WritePrepareFinished {
-        cid,
-        ctx,
-        tag,
-        pr,
-        to_be_write: modifies,
-        rows,
-    }));
+    Ok((ctx, pr, modifies, rows))
+}
 
-    Ok(())
+fn handle_schedule(scheduler: worker::Scheduler<Msg>, msg: Msg) -> bool {
```

Comment:
Maybe send_msg_to_scheduler is a better name. Does this function can be a member of Channels?

Reply:
Or notify_scheduler?